### ISSUE TITLE
Usage of GNU extensions in `lv_draw_sw_letter.c`

### ISSUE THREAD

**Issue by Smertig** [kisvegabor/lvgl#5666]
### LVGL version

master (09ff66b3a457ea2fc7eccf01130e5fac4f58e5d4)

### What happened?

I found usage of [Case Ranges](https://gcc.gnu.org/onlinedocs/gcc/extensions-to-the-c-language-family/case-ranges.html) that's a GNU extension:
https://github.com/lvgl/lvgl/blob/09ff66b3a457ea2fc7eccf01130e5fac4f58e5d4/src/draw/sw/lv_draw_sw_letter.c#L82

It's not portable: https://godbolt.org/z/598E1daax

```
<source>:3:9: warning: range expressions in switch statements are non-standard [-Wpedantic]
    3 |         case 1 ... 2: {
```

I believe it should be fixed, because `lvgl` positions itself as:
> A fully portable C (C++ compatible) library with no external dependencies.

**Comment by and-elf** [comment]
Very interesting! I was sure the compiler flags explicitly removed extensions.. it's that no longer true?

**Comment by gvanem** [comment]
Saw this compile-error on this file using MSVC. No CI for MSVC!?

**Comment by C47D** [comment]
> Saw this compile-error on this file using MSVC. No CI for MSVC!?

I think no CI for MSVC at the moment.

**Comment by kisvegabor** [comment]
I fixed it here: #5698

> Saw this compile-error on this file using MSVC. No CI for MSVC!?

Is it possible? :thinking: I don't really know Windows tool.

**Comment by kisvegabor** [comment]
After 5 second of research: https://github.com/danielzsh/spark/blob/master/.github/workflows/compile.yml

**Comment by MouriNaruto** [comment]
> I fixed it here: #5698
> 
> > Saw this compile-error on this file using MSVC. No CI for MSVC!?
> 
> Is it possible? 🤔 I don't really know Windows tools.

It's possible. LVGL Visual Studio edition uses GitHub Action CI.

Kenji Mouri
